Dash api

biwano opened the pull request:
Dash-API application

Fixes #154 
Fixes #157
Fixes #159 
Fixes #160
Fixes #161 
Fixes #162 

- [x] Raw Credits endpoint
- [x] Credits aggregations endpoints
  - [x] Total
  - [x] Daily aggregation (normal and cumulative)
  - [x] Monthly aggregation (normal and cumulative)
  - [x] Countries aggregation
  - [x] Projects aggregation
  - [x] Methodologies aggregation
 - [x] Raw pools endpoint
 - [x] Pools aggregations endpoint
  - [x] Total
  - [x] Daily aggregation (normal and cumulative)
  - [x] Monthly aggregation (normal and cumulative)
- [x] Credits Holders endpoint
- [x] Token prices endpoints
  - [x] Latest Price
  - [x] History
- [x] Carbon Supply endpoint
- [x] Raw Retirements endpoint
- [x] Retirements aggregations endpoints
  - [x] Pools aggregations
  - [x] Tokens aggregations
  - [x] Chains aggregations
  - [x] Beneficiaries aggregations

0xAurelius commented on src/apps/services/cache.py:
```diff
@@ -0,0 +1,281 @@
+                .dt.floor("D")
+                .dt.date
+            )
+            """
```
What's the deal with this docstring comment? Unnecessary right now? More functionality to be added later?

Should be a `TODO:` on it if so...

biwano replied:
Ho no my bad. I just forgot to remove it.

0xAurelius reviewed (changes requested):
Raised one minor issue with a stray docstring commented chunk of code

It does seem like we could rename the API endpoints to refer to `credits` instead of `offsets` and then later on deal with the change in the underlying subgraphs... that way when we build the new frontend that queries the API, it will already be referring to `credits` instead of `offsets` (even though the underlying datasets are still called `offsets`)

AFAICT none of the actual response contents use the term `offset` - only the endpoint names...

biwano commented:
> Raised one minor issue with a stray docstring commented chunk of code
> 
> It does seem like we could rename the API endpoints to refer to `credits` instead of `offsets` and then later on deal with the change in the underlying subgraphs... that way when we build the new frontend that queries the API, it will already be referring to `credits` instead of `offsets` (even though the underlying datasets are still called `offsets`)
> 
> AFAICT none of the actual response contents use the term `offset` - only the endpoint names...

I changed the endpoints and refactored the code
